components/Results.js

```diff
@@ -46,26 +47,55 @@ export default class Results extends React.Component {
           </View>
         );
       } else {
-          console.log("There are ..." + this.state.dataSource.product.ingredients_from_palm_oil_tags.length())
+          // console.log(this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_tags )
           //Problem here - the ingredients_from_palm_oil_n field doesn't exist on all records. It seems that ngredients_from_palm_oil_tags works though!
-          if(this.state.dataSource.product.ingredients_from_palm_oil_n == 0){
+          if (this.state.dataSource.product.ingredients_from_palm_oil_n > 0 || (this.state.dataSource.product.ingredients_from_palm_oil_tags != undefined && this.state.dataSource.product.ingredients_from_palm_oil_tags.length > 0)) {
```

Comment:
Before merging, could we refactor by creating constants like: 

-   const palm_ingredients_total = this.state.dataSource.product.ingredients_from_palm_oil_n;
-         const palm_ingredients_list = this.state.dataSource.product.ingredients_from_palm_oil_tags;
-         const maybe_total = this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_n;
-         const maybe_list = this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_tags;

I currently have it on my computer like this